src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1488,6 +1488,63 @@ void addIfNotNull(INamedTypeSymbol? toAdd)
             ImmutableArray<bool> memberIsReadOnly,
             ImmutableArray<NullableAnnotation> memberNullableAnnotations);
 
+        /// <summary>
+        /// Creates an <see cref="IMethodSymbol"/> whose <see cref="IMethodSymbol.MethodKind"/> is <see
+        /// cref="MethodKind.BuiltinOperator"/> for a binary operator. Built-in operators are commonly created for
+        /// symbols like <c>bool int.operator==(int v1, int v2)</c> which the language implicitly supports, even if such
+        /// a symbol is not explicitly defined for that type in either source or metadata.
+        /// </summary>
+        /// <param name="name">The binary operator name.  Should be one of the names from <see cref="WellKnownMemberNames"/>.</param>
+        /// <param name="returnType">The return type of the binary operator.</param>
+        /// <param name="leftType">The type of the left operand of the binary operator.</param>
+        /// <param name="rightType">The type of the right operand of the binary operator.</param>
+        /// <param name="isChecked">Whether or not this is a checked binary operator.  For example <c>int int.operator
+        /// checked +(int v1, int v2)</c>.</param>
+        /// <returns></returns>
+        public IMethodSymbol CreateBuiltinOperator(string name, ITypeSymbol returnType, ITypeSymbol leftType, ITypeSymbol rightType, bool isChecked)
```

Comment:
The signature uses a string parameter, while the referenced issue with approved API uses `BinaryOperatorKind`. Is that intentional?

Reply:
yes.  going down this path i found that that we needed to map the enum to an enum just to get this string.  and the enums don't contain all actual values either (like the names for increment operators . bypassing this was just simpler (esp. as the underlying type only uses the name anyways).